File: app/src/main/java/com/zerostress/manager/SendNotificationActivity.java
package com.zerostress.manager;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.button.MaterialButton;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class SendNotificationActivity extends AppCompatActivity {

    private EditText etTitle, etMessage;
    private ProgressBar progressBar;
    private FirebaseFirestore db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_send_notification);

        db = FirebaseFirestore.getInstance();
        etTitle = findViewById(R.id.etNotificationTitle);
        etMessage = findViewById(R.id.etNotificationMessage);
        progressBar = findViewById(R.id.progressBar);
        MaterialButton btnSend = findViewById(R.id.btnSendNotification);
        MaterialButton btnQuickAlert = findViewById(R.id.btnQuickAlert);

        btnSend.setOnClickListener(v -> sendNotification());
        btnQuickAlert.setOnClickListener(v -> showQuickAlertDialog());
    }

    private void sendNotification() {
        String title = etTitle.getText().toString().trim();
        String message = etMessage.getText().toString().trim();

        if (TextUtils.isEmpty(title)) {
            etTitle.setError("Enter title");
            return;
        }
        if (TextUtils.isEmpty(message)) {
            etMessage.setError("Enter message");
            return;
        }

        progressBar.setVisibility(View.VISIBLE);

        Map<String, Object> notification = new HashMap<>();
        notification.put("title", title);
        notification.put("message", message);
        notification.put("type", "admin");
        notification.put("timestamp", System.currentTimeMillis());
        notification.put("sentBy", "Admin");

        // Save to Firestore (players listening in-dashboard will see it)
        db.collection("notifications").add(notification)
            .addOnSuccessListener(docRef -> {
                // Also save with ID for tracking
                docRef.update("id", docRef.getId());
                progressBar.setVisibility(View.GONE);
                Toast.makeText(this, "✅ Notification sent to all players!\n" +
                    "Players will see it in-app and via push.", Toast.LENGTH_LONG).show();
                etTitle.setText("");
                etMessage.setText("");
            })
            .addOnFailureListener(e -> {
                progressBar.setVisibility(View.GONE);
                Toast.makeText(this, "Failed: " + e.getMessage(), Toast.LENGTH_SHORT).show();
            });
    }

    private void showQuickAlertDialog() {
        String[] quickAlerts = {
            "Match starting in 5 minutes!",
            "Tournament begins now!",
            "Server maintenance in 10 minutes",
            "Double XP event active!",
            "New season coming soon!"
        };

        new AlertDialog.Builder(this)
            .setTitle("Quick Alerts")
            .setItems(quickAlerts, (dialog, which) -> {
                etTitle.setText("📢 Quick Alert");
                etMessage.setText(quickAlerts[which]);
                sendNotification();
            })
            .show();
    }
}

File: app/src/main/java/com/zerostress/manager/SendNotificationActivity.kt
package com.zerostress.manager

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.messaging.FirebaseMessaging
import com.zerostress.manager.ui.theme.*

class SendNotificationActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent { ZeroStressTheme { SendNotificationScreen() } }
    }
}

@Composable
fun SendNotificationScreen() {
    var title by remember { mutableStateOf("") }
    var message by remember { mutableStateOf("") }
    var status by remember { mutableStateOf("") }

    LaunchedEffect(Unit) {
        FirebaseMessaging.getInstance().subscribeToTopic("all_players")
        FirebaseMessaging.getInstance().subscribeToTopic("match_updates")
        FirebaseMessaging.getInstance().subscribeToTopic("announcements")
    }

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("🔔 Send Notification", fontSize = 22.sp, fontWeight = FontWeight.Bold, color = Color.White)
        Spacer(modifier = Modifier.height(16.dp))

        Card(shape = RoundedCornerShape(16.dp), colors = CardDefaults.cardColors(containerColor = ZSCard)) {
            Column(modifier = Modifier.padding(20.dp)) {
                OutlinedTextField(value = title, onValueChange = { title = it }, label = { Text("Notification Title") }, modifier = Modifier.fillMaxWidth(), singleLine = true)
                Spacer(modifier = Modifier.height(8.dp))
                OutlinedTextField(value = message, onValueChange = { message = it }, label = { Text("Notification Message") }, modifier = Modifier.fillMaxWidth(), minLines = 3)
                Spacer(modifier = Modifier.height(16.dp))

                Button(
                    onClick = {
                        if (title.isNotBlank() && message.isNotBlank()) {
                            FirebaseFirestore.getInstance().collection("notifications").add(
                                mapOf(
                                    "title" to title.trim(),
                                    "message" to message.trim(),
                                    "author" to "Admin",
                                    "timestamp" to System.currentTimeMillis(),
                                    "topic" to "all_players"
                                )
                            ).addOnSuccessListener {
                                status = "✅ Notification sent!"
                                title = ""
                                message = ""
                            }
                        }
                    },
                    modifier = Modifier.fillMaxWidth()
                ) { Text("Send to All Players") }

                Spacer(modifier = Modifier.height(8.dp))
                OutlinedButton(
                    onClick = {
                        val quickAlerts = listOf(
                            "Maintenance" to "Server maintenance in 30 minutes.",
                            "Update" to "New update available! Check the changelog.",
                            "Event" to "Special event starting now!"
                        )
                        quickAlerts.random().let { (t, m) ->
                            title = t
                            message = m
                        }
                    },
                    modifier = Modifier.fillMaxWidth()
                ) { Text("Quick Alert") }

                if (status.isNotEmpty()) {
                    Spacer(modifier = Modifier.height(12.dp))
                    Text(status, color = ZSPrimary)
                }
            }
        }
    }
}
